CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+cmake_minimum_required(VERSION 3.9)
+
+project(uap-cpp)
+
+IF(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+    SET(CMAKE_BUILD_TYPE Release)
+ENDIF(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -Werror -fPIC")
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0")
+
+
+
+set(LIB_SOURCES
+        UaParser.cpp
+        internal/AlternativeExpander.cpp
+        internal/Pattern.cpp
+        internal/ReplaceTemplate.cpp
+        internal/SnippetIndex.cpp)
+
+find_package(yaml-cpp REQUIRED)
+include_directories(${YAML_CPP_INCLUDE_DIR})
+#link_directories(${YAML_CPP_LIBRARY})
```

Comment:
Why is this commented out?

Reply:
Oh, sorry, removed. First it should be `YAML_CPP_LIBRARIES`, second they are broken in both my test envs (Ubuntu and FreeBSD), because they just resolve to `yaml-cpp` instead of a full path.
Everything compiles without it anyway, so I've just removed it.